src/emulator/cpu.c

```diff
@@ -1,7580 +1,7743 @@
+    if (eCode == CEC_NONE) {
+        return false;
     }
-
-    if((ex >= CP0_EX_RFU16 && ex <= CP0_EX_RFU22) || (ex >= CP0_EX_RFU24 && ex <= CP0_EX_RFU30)) {
-        return 0;
+    if ((eCode >= CEC_RESERVED_16 && eCode <= CEC_RESERVED_22) ||
+        (eCode >= CEC_RESERVED_24 && eCode <= CEC_RESERVED_30)) {
+        return false;
     }
-
-    if(ex == CP0_EX_RSVD_INS) {
-        return 0;
+    if (eCode == CEC_RESERVED) {
+        return false;
     }
 
-    if(ex == CP0_EX_INTERRUPT) {
-        if(!cpuTestInterruptInl(cpu, intr)) {
-            return 0;
+    if (eCode == CEC_INTERRUPT) {
+        if (!cpuTestInterrupt(pCPU, nMaskIP)) {
+            return false;
         }
     } else {
-        cpu->pc -= 4;
-        cpu->status |= 4;
+        pCPU->nPC -= 4;
+        pCPU->nMode |= 4;
     }
 
-    cpu->status &= ~8;
-    if(!(cpu->status & 0x10)) {
-        cpuHackHandler(cpu);
-        cpu->status |= 0x10;
+    pCPU->nMode &= ~8;
+    if (!(pCPU->nMode & 0x10)) {
+        if (!cpuHackHandler(pCPU)) {}
+        pCPU->nMode |= 0x10;
     }
-
-    if(cpu->unk_0x24 != -1) {
-        cpu->unk_0x24 = -1;
-        cpu->cp0[CP0_EPC].d = cpu->pc - 4;
-        cpu->cp0[CP0_CAUSE].d |= CAUSE_BD;
+    if (pCPU->nWaitPC != 0xFFFFFFFF) {
+        pCPU->nWaitPC = -1;
+        pCPU->anCP0[14] = pCPU->nPC - 4;
+        pCPU->anCP0[13] |= 0x80000000;
     } else {
-        cpu->cp0[CP0_EPC].d = cpu->pc;
+        pCPU->anCP0[14] = pCPU->nPC;
     }
 
-    cpu->status &= ~0x80;
-    cpu->cp0[CP0_STATUS].w[1] |= 2;
-    cpu->cp0[CP0_CAUSE].sd = (cpu->cp0[CP0_CAUSE].sd & ~0x7C) | (ex << 2);
+    pCPU->nMode &= ~0x80;
+    pCPU->anCP0[12] |= 2;
+    pCPU->anCP0[13] = (pCPU->anCP0[13] & ~0x7C) | (eCode << 2);
 
-    if(ex < 4) {
-        cpu->pc = 0x80000000;
+    if (eCode - 1 <= 2U) {
+        pCPU->nPC = 0x80000000;
     } else {
-        cpu->pc = 0x80000180;
+        pCPU->nPC = 0x80000180;
     }
 
-    cpu->status |= 0x24;
+    pCPU->nMode |= 4;
+    pCPU->nMode |= 0x20;
 
-    if(!func_8005D614(gSystem->unk_0x0058, cpu, -1)) {
-        return 0;
+    if (!libraryCall(SYSTEM_LIBRARY(gpSystem), pCPU, -1)) {
+        return false;
     }
 
-    return 1;
+    return true;
 }
-#else
-s32 cpuException(cpu_class_t *cpu, s32 ex, u32 intr);
-#pragma GLOBAL_ASM("asm/non_matchings/virtual_console/cpu/cpuException.s")
-#endif
 
-/**
- * @brief Creates a new device and registers memory space for that device.
- * 
- * @param cpu The emulated VR4300.
- * @param new_dev_idx A pointer to the index in the cpu->devices array which the device was created.
- * @param dev_obj The object which will handle reuqests for this device.
- * @param vaddr Starting address of the device's address space.
- * @param paddr Starting physical address of the device's address space.
- * @param size Size of the device's memory space.
- * @param create_arg An argument which will be passed back to the device's event handler.
- * @return s32 1 on success, 0 otherwise.
- */
```

Comment:
Should we preserve these comments? Seems kinda sad to lose them, although the param names would need to be updated (and bool return values too)

Reply:
done, doing this made me found a tiny error this file had, some functions were returning ``bool`` when it should be ``s32`` (the ones that are used as a ``CpuExecuteFunc`` pointer)